packages/agreement-process/test/computeAgreementsState.test.ts

```diff
@@ -76,145 +63,176 @@ describe("compute Agreement state by attribute", () => {
-      ...getMockEService(),
-      producerId: generateId(),
-      descriptors: [descriptor2],
-    };
+      const updatableActiveAgreement: Agreement = {
+        ...getMockAgreement(eservice.id, consumer.id, agreementState.active),
+        descriptorId: eservice.descriptors[0].id,
+        producerId: eservice.producerId,
+        suspendedByPlatform: false,
+      };
 
-    const updatableAgreement1: Agreement = {
-      ...getMockAgreement(
-        eservice1.id,
-        consumer.id,
-        randomArrayItem([agreementState.draft, agreementState.pending])
-      ),
-      descriptorId: eservice1.descriptors[0].id,
-      producerId: eservice1.producerId,
-      suspendedByPlatform: false,
-    };
+      await addOneAgreement(updatableActiveAgreement);
 
-    const updatableAgreement2: Agreement = {
-      ...getMockAgreement(eservice2.id, consumer.id, agreementState.active),
-      descriptorId: eservice2.descriptors[0].id,
-      producerId: eservice2.producerId,
-      suspendedByPlatform: false,
-    };
+      await agreementService.computeAgreementsStateByAttribute(
+        invalidCertifiedAttribute.id,
+        consumer,
+        {
+          authData,
+          serviceName: "",
+          correlationId: "",
+          logger: genericLogger,
+        }
+      );
 
-    const nonUpdatableAgreement = {
-      ...getMockAgreement(
-        eservice1.id,
-        consumer.id,
-        randomArrayItem([agreementState.archived, agreementState.rejected])
-      ),
-      descriptorId: eservice1.descriptors[0].id,
-      producerId: eservice1.producerId,
-    };
+      const agreementStateUpdateEvent = await readLastAgreementEvent(
+        updatableActiveAgreement.id
+      );
 
-    await addOneEService(eservice1);
-    await addOneEService(eservice2);
-    await addOneAgreement(updatableAgreement1);
-    await addOneAgreement(updatableAgreement2);
-    await addOneAgreement(nonUpdatableAgreement);
+      expect(agreementStateUpdateEvent).toMatchObject({
+        type: "AgreementSuspendedByPlatform",
+        event_version: 2,
+        version: "1",
+        stream_id: updatableActiveAgreement.id,
+      });
+
+      const agreementStateUpdateEventData = decodeProtobufPayload({
+        messageType: AgreementSuspendedByPlatformV2,
+        payload: agreementStateUpdateEvent.data,
+      });
+
+      expect(agreementStateUpdateEventData).toMatchObject({
+        agreement: toAgreementV2({
+          ...updatableActiveAgreement,
+          state: agreementState.suspended,
+          suspendedByPlatform: true,
+        }),
+      });
+    });
 
-    await agreementService.computeAgreementsStateByAttribute(
-      invalidCertifiedAttribute.id,
-      consumer,
-      {
-        authData,
-        serviceName: "",
-        correlationId: "",
-        logger: genericLogger,
-      }
-    );
+    it("updates the state of an updatable Agreement from Draft or Pending to MissingCertifiedAttributes", async () => {
```

Comment:
may I ask to split this test too? 😅 
or even better, we could use [this](https://github.com/pagopa/interop-be-monorepo/blob/main/packages/purpose-process/test/testArchivePurposeVersion.ts#L222).

if we have a finite number of cases I prefer to test them all instead of using the random approach

Reply:
Nice! 🫶 I refactored it using `it.each`